x/gamm/keeper/pool_service.go

```diff
@@ -177,6 +177,9 @@ func (k Keeper) CreatePool(ctx sdk.Context, msg types.CreatePoolMsg) (uint64, er
 // JoinPoolNoSwap determines the maximum amount that can be LP'd without any swap,
 // by looking at the ratio of the total LP'd assets. (e.g. 2 osmo : 1 atom)
 // It then finds the maximal amount that can be LP'd.
+//
+// All pool interfaces handled within this method are pointer references, `JoinPool` directly updates the pool interface.
```

Comment:
I think it makes sense to have this comment inside the method but not as a spec.

Method specs should not be concerned with the implementation details so that we can treat methods as black boxes